docs/modules/ROOT/pages/usage.adoc

```diff
@@ -0,0 +1,390 @@
+This allows the contract to have indirect access to the component's storage.
+
+The `#[flat]` attribute for events in the `Event` enum, however, is not required.
+Component events are not flattened in the component itself because it would remove the event ID from the event log.
+Note that if contracts do not flatten component events, the first key in the event log will be the component ID.
+By flattening the component event in the contract, the first key will be the event ID.
+
+=== Implementations
+
+:accessing-storage: xref:/usage.adoc#accessing_component_storage[Accessing component storage]
+
+Components come with granular implementations of different interfaces.
+This allows contracts to integrate only the implementations that they'll use and avoid unnecessary bloat.
+Integrating an implementation looks like this:
+
+[,javascript]
+----
+mod MyContract {
+    use openzeppelin::security::InitializableComponent;
+
+    component!(path: InitializableComponent, storage: initializable, event: InitializableEvent);
+
+    (...)
+
+    // Gives the contract access to the implementation methods
+    impl InitializableImpl =
+        InitializableComponent::InitializableImpl<ContractState>;
+}
+----
+
+Defining an `impl` gives the contract access to the methods within the implementation from the component.
+For example, `is_initialized` is defined in the `InitializableImpl`.
+A function on the contract level can expose it like this:
+
+[,javascript]
+----
+#[starknet::contract]
+mod MyContract {
+    use openzeppelin::security::InitializableComponent;
+
+    component!(path: InitializableComponent, storage: initializable, event: InitializableEvent);
+
+    (...)
+
+    impl InitializableImpl =
+        InitializableComponent::InitializableImpl<ContractState>;
+
+    #[external(v0)]
+    fn is_initialized(ref self: ContractState) -> bool {
+        self.initializable.is_initialized()
+    }
+}
+----
+
+Notice that the function must pass the state (`self`) and the component's storage (`initializable`) before finally accessing the implementation method.
+
+NOTE: Contracts can also (indirectly) access a component's storage. See {accessing-storage}.
+
+While there's nothing wrong with manually exposing methods like in the previous example, this process can be tedious for implementations with many methods.
+Fortunately, a contract can embed implementations into the ABI which will expose all of the methods of the implementation.
+To embed an implementation, add the `#[abi(embed_v0)]` attribute above the `impl`:
+
+[,javascript]
+----
+#[starknet::contract]
+mod MyContract {
+    (...)
+
+    // This attribute exposes the methods of the `impl`
+    #[abi(embed_v0)]
+    impl InitializableImpl =
+        InitializableComponent::InitializableImpl<ContractState>;
+}
+----
+
+`InitializableImpl` defines the `is_initialized` method in the component.
+By adding the embed attribute, `is_initialized` becomes a contract entrypoint for `MyContract`.
+
```

Comment:
since this whole section is about explaining how to use the library, I think it's worth mentioning that this concept guides big part of our API design: we separate implementations between embeddable and internal, making it easy to embed safely. We should mention that and maybe also link to some API doc example -- which can be hard to maintain/not break over time 😅

Reply:
Good idea. I added it as a TIP at the end of the `Implementations` section. Let me know what you think